src/client.h

```diff
@@ -499,6 +499,9 @@ class Client : public con::PeerHandler, public InventoryManager, public IGameDef
 	u8 getProtoVersion()
 	{ return m_proto_ver; }
 
+	bool connectedToServer()
```

Comment:
It locks a mutex in the connection in order to execute, so it does modify state...